.github/workflows/ci.yaml

```diff
@@ -64,18 +60,39 @@ jobs:
           && sudo rm -rf /var/cache/apt/* \
           && sudo apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false
 
+      - name: Set up python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Create cache file
+        run: echo '${{ matrix.python-version }}' > ./matrix-file.txt
+
       - uses: pdm-project/setup-pdm@v3
         name: Set up PDM
         with:
           python-version: ${{ matrix.python-version }}
           allow-python-prereleases: true
           cache: true
-
+          cache-dependency-path: |
+            ./pdm.lock
+            ./matrix-file.txt
+      - name: Install base libraries
+        # install a few core libraries for testing and building
+        run: pip install nodeenv cython setuptools mypy cython pip --upgrade --quiet --user
       - name: Install dependencies
-        run: pdm install
-
-      - name: Test
-        run: pdm run pytest -m "" -n auto
+        run: pdm install -G:all
+      - name: Test with Coverage
```

Comment:
Does this need to be run for all python versions?

Reply:
How about something like [this](https://github.com/litestar-org/polyfactory/blob/e1f7a47c6e3c979f8f43ce6ee612f417d2f3c03b/.github/workflows/ci.yaml#L84-L86)? Or the litestar one which also only tests with coverage for 3.12 and pydantic v2 ([reference](https://github.com/litestar-org/litestar/blob/3a2089a96483652f8cadb4292eb9ae42a28f4db6/.github/workflows/ci.yml#L43)). It does have a more elaborate setup than the polyfactory one though.

Reply:
Thanks.  I've updated it to work like the other repos.